 error[E0308]: mismatched types
   --> .../index.crates.io-*/sys_metrics-0.2.6/src/host/sys/linux/users.rs:29:28
    |
 29 |         let cuser: &[i8] = &cbuffer.ut_user;
    |                    -----   ^^^^^^^^^^^^^^^^ expected `&[i8]`, found `&[u8; 32]`
    |                    |
    |                    expected due to this
    |
    = note: expected reference `&[i8]`
               found reference `&[u8; 32]`

 For more information about this error, try `rustc --explain E0308`.
 error: could not compile `sys_metrics` (lib) due to previous error

 --

--- a/src/host/sys/linux/users.rs
+++ b/src/host/sys/linux/users.rs
@@ -26,7 +26,7 @@ pub fn get_logged_users() -> Result<Vec<
     } != 0
     {
         let cbuffer = unsafe { buffer.assume_init() };
-        let cuser: &[i8] = &cbuffer.ut_user;
+        let cuser: &[u8; 32] = &cbuffer.ut_user;
 
         if cuser[0] != 0 && cbuffer.ut_type == libc::USER_PROCESS {
             let csuser = to_str(cuser.as_ptr().cast()).trim_matches('\0').to_owned();
